File: test_commits/javascript/refactor_original5.js
const jwt = require("jsonwebtoken");

function handleLogin(user, password, loginAttempts, secret) {
    if (user.password !== password) {
        loginAttempts[user.username] = (loginAttempts[user.username] || 0) + 1;
        if (loginAttempts[user.username] > 5) {
            console.log("Account locked due to too many failed attempts");
        } else {
            console.log("Incorrect password");
        }
        return null;
    }

    loginAttempts[user.username] = 0;
    user.lastLogin = Date.now();
    console.log(`${user.username} logged in at ${user.lastLogin}`);

    const payload = {
        userId: user.id,
        username: user.username,
        iat: Math.floor(Date.now() / 1000)
    };
    return jwt.sign(payload, secret);
}

File: test_commits/java/refactor_original5.java
import java.time.Instant;
import java.util.Map;

public class AuthService {

    public static String handleLogin(User user, String password, Map<String, Integer> loginAttempts, String secret) {
        if (!user.getPassword().equals(password)) {
            loginAttempts.put(user.getUsername(), loginAttempts.getOrDefault(user.getUsername(), 0) + 1);
            if (loginAttempts.get(user.getUsername()) > 5) {
                System.out.println("Account locked due to too many failed attempts");
            } else {
                System.out.println("Incorrect password");
            }
            return null;
        }

        loginAttempts.put(user.getUsername(), 0);
        user.setLastLogin(Instant.now().toEpochMilli());
        System.out.println(user.getUsername() + " logged in at " + user.getLastLogin());

        String payload = "{\"user_id\":\"" + user.getId() + "\",\"username\":\"" + user.getUsername() + "\",\"iat\":" + Instant.now().getEpochSecond() + "}";
        return JwtUtils.sign(payload, secret);
    }
}
